safe-global/web-core
# fix: display spinner when proposing signature

## What it solves

Signatures not immediately visible.

## How this PR fixes it

Signature proposals now poll for queued transaction. However, the immediately poll does not always return the proposed signature. As such, a new hook `useIsSignatureProposalPending` also subscribed to signature proposals and calculates if a proposed signature is pending. It is used as a loading flag.

## How to test it

Create a transaction on a 2+/n Safe. Switch owner and sign the transaction. Observe the loading spinner and disabled confirmation button.

## Screenshots
![sign](https://user-images.githubusercontent.com/20442784/185977807-6a073091-994f-44a0-afa3-9c33f963975b.gif)
